src/messaging/ExchangeContext.cpp

```diff
@@ -537,6 +537,10 @@ CHIP_ERROR ExchangeContext::HandleMessage(uint32_t messageCounter, const Payload
     // layer has completed its work on the ExchangeContext.
     ExchangeHandle ref(*this);
 
+    // If we received a message from the Peer and we are still expecting a response, we know the peer is active
+    // If we receive a message from the Peer but we are not expecting a response, we don't know if the peer is still active
```

Comment:
I'm not quite following this logic.

If _we_ are expecting a response, why does that have anything to do with whether our _peer_ is active?

Reply:
The goal is to determine if we re-send the message at a faster rate if we know the receiver ie. our peer is "active".
In this case, if we receive a message, we assume, if we were expecting a response the peer is active and we remove the boolean once we determine if it is the response.

Setting it to true here covers the case were we receive the duplicate before the initial message or if we receive a standalone ack before the expected response.

Reply:
> if we were expecting a response the peer is active

This is the part that does not make sense to me.  We just got a message from the peer.  Why does it matter whether _we_ were expecting a response (which would make _us_ active), for purposes of determining whether the _peer_ is active.

Reply:
The idea is, if we expect a response for our last message but only received an ack for it, that means the peer is active since it still needs to send a message over the existing exchange i.g. the response.

Reply:
OK, that part is _very_ unclear from the comments, and should be documented more clearly.

But what ensures that the peer will in fact remain active?  What is the spec status here?  Does it require things to stay active while they are waiting for a message, or while there is any exchange at all?

Concretely, say the peer acks our message and knows it will take it 20s to perform the command and respond.  Is it allowed to shut down the radio while it's doing that, per spec?

Reply:
Spec says as long as an ExchangeContext is open it will be reachable. For an ICD (which is the case where the radio might be off), it will be fast polling during the full 20s of operation.